src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -200,7 +227,13 @@ public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResou
             // an event hub namespace without an event hub? :)
             if (builder.Resource.Hubs is { Count: > 0 } && builder.Resource.Hubs[0] is { } hub)
             {
-                var healthCheckConnectionString = $"{connectionString};EntityPath={hub.Name};";
+                // Endpoint=... format
+                var props = EventHubsConnectionStringProperties.Parse(connectionString);
```

Comment:
This is a pretty heavy way to check the string.  Since `EntityName` is a well-known token in this scenario, it may make sense to consider an `IndexOf` check instead.

Reply:
It's also possible that this code has been written because I suggested to use it 🤷‍♂️. "Heavy" doesn't matter when we try to run the Event Hubs emulator or provision it ...

Reply:
> It's also possible that this code has been written because I suggested to use it 🤷‍♂️. "Heavy" doesn't matter when we try to run the Event Hubs emulator or provision it ...

What he said :)